io/src/main/scala/fs2/io/file/Files.scala

```diff
@@ -0,0 +1,510 @@
+    ): Stream[F, Path] =
+      Stream.resource(tempDirectoryResource(dir, prefix, attributes))
+
+    def tempDirectoryResource(
+        dir: Path,
+        prefix: String,
+        attributes: Seq[FileAttribute[_]]
+    ): Resource[F, Path] =
+      Resource.make {
+        Sync[F].blocking(JFiles.createTempDirectory(dir, prefix, attributes: _*))
+      } { p =>
+        deleteDirectoryRecursively(p)
+          .recover { case _: NoSuchFileException => () }
+      }
+
+    def walk(start: Path): Stream[F, Path] =
+      walk(start, Seq.empty)
+
+    def walk(start: Path, options: Seq[FileVisitOption]): Stream[F, Path] =
+      walk(start, Int.MaxValue, options)
+
+    def walk(start: Path, maxDepth: Int, options: Seq[FileVisitOption]): Stream[F, Path] =
+      _runJavaCollectionResource[JStream[Path]](
+        Sync[F].blocking(JFiles.walk(start, maxDepth, options: _*)),
+        _.iterator.asScala
+      )
+
+    def writeAll(
+        path: Path,
+        flags: Seq[StandardOpenOption] = List(StandardOpenOption.CREATE)
+    ): Pipe[F, Byte, INothing] =
+      in =>
+        Stream
+          .resource(WriteCursor.fromPath(path, flags))
+          .flatMap(_.writeAll(in).void.stream)
+
+  }
+}
+
+trait Files[F[_]] extends BaseFiles[F] {
+
+  /**
+    * Returns an infinite stream of data from the file at the specified path.
+    * Starts reading from the specified offset and upon reaching the end of the file,
+    * polls every `pollDuration` for additional updates to the file.
+    *
+    * Read operations are limited to emitting chunks of the specified chunk size
+    * but smaller chunks may occur.
+    *
+    * If an error occurs while reading from the file, the overall stream fails.
+    */
+  def tail(
+      path: Path,
+      chunkSize: Int,
+      offset: Long = 0L,
+      pollDelay: FiniteDuration = 1.second
+  ): Stream[F, Byte]
+
+  /**
+    * Creates a [[Watcher]] for the default file system.
+    *
+    * The watcher is returned as a resource. To use the watcher, lift the resource to a stream,
+    * watch or register 1 or more paths, and then return `watcher.events()`.
+    */
+  def watcher: Resource[F, Watcher[F]]
+
+  /**
+    * Watches a single path.
+    *
+    * Alias for creating a watcher and watching the supplied path, releasing the watcher when the resulting stream is finalized.
+    */
+  def watch(
+      path: Path,
+      types: Seq[Watcher.EventType] = Nil,
+      modifiers: Seq[WatchEvent.Modifier] = Nil,
+      pollTimeout: FiniteDuration = 1.second
+  ): Stream[F, Watcher.Event]
+
+  /**
+    * Writes all data to a sequence of files, each limited in size to `limit`.
+    *
+    * The `computePath` operation is used to compute the path of the first file
+    * and every subsequent file. Typically, the next file should be determined
+    * by analyzing the current state of the filesystem -- e.g., by looking at all
+    * files in a directory and generating a unique name.
+    */
+  def writeRotate(
```

Comment:
Possible out of scope for this PR, but there was a question about how to ensure that when using `writeRotate` with strings, of how to align chunks, for example so that printing a single line definitely goes into either one file or another, and wouldn't split in the middle - for example for logging. If that's easy, might be good to get it in here

Reply:
Yeah, I remember seeing that conversation. Maybe one option is defining how `writeRotate` handles a chunk that spans the limit. E.g., given a rotation limit of 1024 and file position of 1000, receiving a chunk of 32 bytes currently writes 24 bytes, rotates, and then 8 more to new file. Maybe that could be configurable as to whether chunks get split -- e.g. a hard limit vs a soft limit.